content/guides/frameworks/laravel/development-setup.md

```diff
@@ -0,0 +1,342 @@
+    echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> /home/www/.bashrc && \
+    echo '[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"' >> /home/www/.bashrc
+
+# Set the working directory
+WORKDIR /var/www
+
+# Override the entrypoint to avoid the default php entrypoint
+ENTRYPOINT []
+
+# Default command to keep the container running
+CMD ["bash"]
+```
+
+### Create Docker Compose Configuration for Development
+
+Here's the `compose.yaml` file to set up the development environment:
+
+```yaml
+services:
+  web:
+    image: nginx:latest # Using the default Nginx image with custom configuration.
+    volumes:
+      # Mount the application code for live updates
+      - ./:/var/www
+      # Mount the Nginx configuration file
+      - ./docker/nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+    ports:
+      # Map port 80 inside the container to the port specified by 'NGINX_PORT' on the host machine
+      - "${NGINX_PORT:-80}:80"
+    environment:
+      - NGINX_HOST=${NGINX_HOST}
+    networks:
+      - laravel
+    depends_on:
+      php-fpm:
+        condition: service_started  # Wait for php-fpm to start
+
+  php-fpm:
+    # For the php-fpm service, we will create a custom image to install the necessary PHP extensions and setup proper permissions.
+    build:
+      context: .
+      dockerfile: ./docker/php-fpm/Dockerfile
+      args:
+        UID: ${UID}
+        GID: ${GID}
+        XDEBUG_ENABLED: ${XDEBUG_ENABLED}
+        XDEBUG_MODE: ${XDEBUG_MODE}
+        XDEBUG_HOST: ${XDEBUG_HOST}
+        XDEBUG_IDE_KEY: ${XDEBUG_IDE_KEY}
+        XDEBUG_LOG: ${XDEBUG_LOG}
+        XDEBUG_LOG_LEVEL: ${XDEBUG_LOG_LEVEL}
+    env_file:
+      # Load the environment variables from the Laravel application
+      - .env
+    user: "${UID}:${GID}"
+    volumes:
+      # Mount the application code for live updates
+      - ./:/var/www
+    networks:
+      - laravel
+    depends_on:
+      postgres:
+        condition: service_started  # Wait for postgres to start
+
+  workspace:
+   # For the workspace service, we will also create a custom image to install and setup all the necessary stuff.
+    build:
+      context: .
+      dockerfile: ./docker/workspace/Dockerfile
+      args:
+        UID: ${UID}
+        GID: ${GID}
+        XDEBUG_ENABLED: ${XDEBUG_ENABLED}
+        XDEBUG_MODE: ${XDEBUG_MODE}
+        XDEBUG_HOST: ${XDEBUG_HOST}
+        XDEBUG_IDE_KEY: ${XDEBUG_IDE_KEY}
+        XDEBUG_LOG: ${XDEBUG_LOG}
+        XDEBUG_LOG_LEVEL: ${XDEBUG_LOG_LEVEL}
+    tty: true  # Enables an interactive terminal
+    stdin_open: true  # Keeps standard input open for 'docker exec'
+    env_file:
+      - .env
+    volumes:
+      - ./:/var/www
+    networks:
+      - laravel
+
+  postgres:
+    image: postgres:16
+    ports:
+      - "${POSTGRES_PORT}:5432"
+    environment:
+      - POSTGRES_DB=${POSTGRES_DATABASE}
+      - POSTGRES_USER=${POSTGRES_USERNAME}
+      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
```

Comment:
This compose file has a lot of dependencies to environment variables. I think it's better to try and keep non-secret values in the compose file directly. Still, the `.env` will need some explanation.